docs/testing/contract-tests/index.md

```diff
@@ -0,0 +1,7 @@
+# Contract Tests
+
+This is where all contract test related documentation is stored.
+
+- [Overview](./testing/contract-tests/contract-tests.md)
```

Comment:
Similar comment to https://github.com/mozilla-services/merino-py/pull/421#discussion_r1327281765 , I'm wondering if we can reduce one level by putting the overview in this document instead.